src/DataViews/LiveFunctionsDataView.cpp

```diff
@@ -272,6 +272,68 @@ std::vector<std::string> LiveFunctionsDataView::GetContextMenu(
   return menu;
 }
 
+std::vector<std::vector<std::string>> LiveFunctionsDataView::GetContextMenuWithGrouping(
+    int clicked_index, const std::vector<int>& selected_indices) {
+  bool enable_select = false;
+  bool enable_unselect = false;
+  bool enable_disassembly = false;
+  bool enable_source_code = false;
+  bool enable_enable_frame_track = false;
+  bool enable_disable_frame_track = false;
+  bool enable_iterator = false;
+
+  const CaptureData& capture_data = app_->GetCaptureData();
+  for (int index : selected_indices) {
+    uint64_t instrumented_function_id = GetInstrumentedFunctionId(index);
+    const FunctionInfo& instrumented_function = GetInstrumentedFunction(index);
+
+    if (app_->IsCaptureConnected(capture_data)) {
+      enable_select |= !app_->IsFunctionSelected(instrumented_function);
+      enable_unselect |= app_->IsFunctionSelected(instrumented_function);
+      enable_disassembly = true;
+      enable_source_code = true;
+    }
+
+    const FunctionStats& stats = capture_data.GetFunctionStatsOrDefault(instrumented_function_id);
+    // We need at least one function call to a function so that adding iterators makes sense.
+    enable_iterator |= stats.count() > 0;
+
+    if (app_->IsCaptureConnected(capture_data)) {
+      enable_enable_frame_track |= !app_->IsFrameTrackEnabled(instrumented_function);
+      enable_disable_frame_track |= app_->IsFrameTrackEnabled(instrumented_function);
+    } else {
+      enable_enable_frame_track |= !capture_data.IsFrameTrackEnabled(instrumented_function_id);
+      enable_disable_frame_track |= capture_data.IsFrameTrackEnabled(instrumented_function_id);
+    }
+  }
+
+  std::vector<std::string> action_group;
+  if (enable_select) action_group.emplace_back(kMenuActionSelect);
+  if (enable_unselect) action_group.emplace_back(kMenuActionUnselect);
+  if (enable_disassembly) action_group.emplace_back(kMenuActionDisassembly);
+  if (enable_source_code) action_group.emplace_back(kMenuActionSourceCode);
+  if (enable_enable_frame_track) action_group.emplace_back(kMenuActionEnableFrameTrack);
+  if (enable_disable_frame_track) action_group.emplace_back(kMenuActionDisableFrameTrack);
+  if (enable_iterator) action_group.emplace_back(kMenuActionIterate);
```

Comment:
Maybe the iterator should be a new group? Just an idea

Reply:
Should I put the iterator and the jump to actions in a same group?